services/gps2tsdb/healthcheck.py

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/python3
+import postgres
+import sys
+import os
+import datetime
+
+threshold = 5
+
+connectionurl='postgresql://' + os.environ['db_user'] + ':' + os.environ['db_password'] + '@postgres:' + os.environ['db_port'] + '/' + os.environ['db_database']
+db = postgres.Postgres(url=connectionurl)
+
+rst = db.one("SELECT * FROM gps ORDER BY time DESC LIMIT 1;")
+
+if rst == None:
+    print("Database has the wrong schema or no data");
+    sys.exit(1)
+
+lastupdate = rst[0]
+tdelta = datetime.datetime.now(lastupdate.tzinfo) - lastupdate
+sdelta = tdelta.total_seconds()
+if sdelta > threshold:
+  print('Database was last updated', sdelta, 'seconds ago, something is wrong ( threshold:', threshold, 's )');
```

Comment:
Would it be possible to make systemd restart gpsd at this point? Currently we are fixing this issue by restarting gpsd from systemd. If we automated this, we could reduce downtime.

Reply:
We would have to change the container to give it access to the host's system systemd. Definitely possible, but I do not believe we have do it before and it breaks down another piece of the `sandbox` idea of containers

Reply:
When we move `gpsd` to a container it could have a similar healthcheck which would restart itself